### DISCUSSION TITLE
Switch between representations on Point() object

### DISCUSSION THREAD

**Discussion by jkissi** [marcomusy/vedo#534] [Q&A] [answered]
Hi Marco, hope you are well! 

I would like to be able switch between views of my pointcloud coloured for my classification or curvature (with the curvature option having an accompanying `cmap.scalarbar()`). 

I also want to be able to turn the normals of the points on or off. You gave me a previous answer using a button to call `plt.remove(pcl).add(new_pcl)` but I am confirming because if I use this method, I would have to have a different point cloud for each of the different combinations: 
```
vpts0 # classification with normals 
vpts1 # classification without normals 
vpts2 # curvature with normals etc.... 
```

Is this the most efficient way of doing this or can I use the `pointdata.select()` method to switch between different point representations on the same cloud?

If I use the `pointdata.select()` method, is there a way to remove the `scalarbar()` from the Points object dynamically and render, or is it the case that once you add this to an object, you're stuck with it? (so you have create a different object if you don't want it to show)

Thanks, hope you had a good weekend!
j

**Comment by marcomusy** [comment, marked as answer]
Hi, you can add and remove any item(s) with 
`plotter.remove(some_list).add(some_other_list)`
(probably you want to do this in a callback function)

Another option is to switch on and off an object visibility with:
`myobject.on()`
`myobject.off()`

For the scalarbar you may need to extract it from the associated mesh and handle it like any other vedo object:
`myscalarbar = mymesh.scalarbar`

Give it a try!

**Reply by marcomusy** [reply]
PS: sorry while reading back I think I missed some of the points you raised.. You can use `select()` you just need to add as many arrays to the point cloud as you wish then you can select the relevant one and `render()` the scene (you do this in a callback or button).
Hope this helps,
Marco
